QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question){
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public bool TryUpdateRecord(int id, string answer, out IRecord recordToUpdate){
```

Comment:
Ah yes, when I wrote .Single() I thought about a potential throw here... 
.First would get around a throw happening if there is more than one record with the same ID (which shouldn't happen..) and we are already checking that there is at least one entry anyways, so .First would always return something.

Does that work?